src/spicedb-common/parsers/dsl/dsl.ts

```diff
@@ -504,16 +508,19 @@ const parensExpr: any = Parsimmon.lazy(() =>
     .or(relationReference)
 );
 
-function BINARY_LEFT(operatorsParser: any, nextParser: any) {
+function BINARY_LEFT(operatorsParser: Parser<string>, nextParser: Parser<ParsedBinaryExpression>) {
```

Comment:
This was the only tricky bit. These types still don't work perfectly; it may make sense to refactor the construction of this parser to be more manual and less functional.